dags/common/sftp_service.py

```diff
@@ -41,10 +37,39 @@ def __connect(self):
             raise DirectoryNotFoundException(
                 "Remote directory doesn't exist. Abort connection."
             )
-        conn.chdir(self.dir)
         return conn
 
+    def __enter__(self):
+        self.connection = self.__connect()
+        return self
+
+    def __exit__(self, exc_type, exc_value, tb):
+        if self.connection:
+            self.connection.close()
+        if exc_type is not None:
+            self.logger.error("Error while exiting SFTPService")
+            traceback.print_exception(exc_type, exc_value, tb)
+        return True
```

Comment:
why we have to override this?

Reply:
is just a safety check, if the connection is set, and if yes, close it

Reply:
if we inherit do we need to handle that?

Reply:
I am not using the inheritance anymore.
If we would inherit, I believe no, we don't need to handle it